clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -1034,7 +1034,7 @@ private void close(long timeout, TimeUnit timeUnit, boolean swallowException) {
                     try {
                         this.ioThread.join(timeUnit.toMillis(timeout));
                     } catch (InterruptedException t) {
-                        firstException.compareAndSet(null, t);
+                        firstException.compareAndSet(null, new InterruptException(t));
```

Comment:
I would probably suggest using the raw `InterruptedException` here and throwing it as `InterruptException` below. Then the trace will be accurate.